app/src/editor/dashboard/index.jsx

```diff
@@ -210,9 +211,19 @@ const DashboardEdit = withRouter(class DashboardEdit extends Component {
     }
 })
 
-const DashboardLoader = withRouter(withErrorBoundary(ErrorComponentView)(class DashboardLoader extends React.PureComponent {
+const ErrorComponent = ({ error, ...props }) => {
+    if (error instanceof ResourceNotFoundError) {
```

Comment:
Same here. Bubble up if "Not Found". Let the routing-level error boundary handle it.